### PR TITLE
Make click_rich extra dependency; Migrate tox to nox

### LINKED ISSUE #303: Make rich_click an optional dependency
rich_click has lots of dependencies and ideally I'd like to be able to use robotframework-tidy without those (my use case is vendoring robotframework-tidy in the language server, and it's kind of a small pain to get all those deps for the cli when all that's needed is using the internal api).

Maybe an option could be creating a `robotframework-tidy-core` with the minimum amount of dependencies and then having `robotframework-tidy` with the big dependencies for the cli...

### PR DESCRIPTION
**Pull request by bhirsz**
click_rich introduce a lot of extra dependencies that are not required to run robotidy and bloats install for users who only want to use robotidy API. 

This change move ``click_rich`` dependency to ``rich`` profile. Install robotidy with rich profile to get it:

```
pip install robotframework-tidy[rich]
```

Migrated tox to nox. It's easier to create dev scripts with Python itself and it also allows to easily install the package for testing in a way it's done by user. Reduced the amount of placed with defined requirements list.

There was also typo extras_requires -> extras_require (easy one to make - install_requires have ``s`` and setup.py doesn't complain about it). Our extras profiles didn't work since the beginning!

Closes #303

### CODE REVIEW

File: robotidy/rich_console.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            click.echo(msg)
+
+    class Markdown:
+        def __init__(self, documentation, **kwargs):
+            self.documentation = documentation
+
+        def __str__(self):
+            return self.documentation
+
+    class Table:
```
Comment (bhirsz) [inline robotidy/rich_console.py:28]: If we don't have rich installed we're using this class to create table-like outputs (for example for list of transformers). It only provides the basic output - but doesn't need any extra dependency.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -15,10 +15,18 @@ Create your own fork of Robotidy first (instructions [here](https://docs.github.
...
 from the source code with `dev` profile. Run following in the root of the project:
 ```
-pip install -e .[dev]
+pip install -e .[dev,rich]
 ```
-Option `-e` allows changes done by you to be automatically reloaded without need for reinstalling the package. `dev` profile 
-installs extra libraries useful for testing and scripting (such as pytest or invoke).
+Option `-e` allows changes done by you to be automatically reloaded without need for reinstalling the package.
+`dev` profile installs extra libraries useful for testing and scripting (such as pytest or invoke).
+`rich` profile provide prettified console outputs.
```
Comment (mnojek) [inline CONTRIBUTING.md:22]:
```suggestion
`rich` profiles provides prettified console outputs.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,10 +15,18 @@ Create your own fork of Robotidy first (instructions [here](https://docs.github.
 It will be easier to create Pull Requests later with your own fork. After cloning your fork to your disk install Robotidy
 from the source code with `dev` profile. Run following in the root of the project:
 ```
-pip install -e .[dev]
+pip install -e .[dev,rich]
 ```
-Option `-e` allows changes done by you to be automatically reloaded without need for reinstalling the package. `dev` profile 
-installs extra libraries useful for testing and scripting (such as pytest or invoke).
+Option `-e` allows changes done by you to be automatically reloaded without need for reinstalling the package.
```
Comment (mnojek) [inline CONTRIBUTING.md:20]:
```suggestion
Option `-e/--editable` allows changes done by you to be automatically reloaded without need for reinstalling the package.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -51,6 +59,10 @@ it will remove node from your file.
 See examples of creating external transformer [here](https://robotidy.readthedocs.io/en/latest/external_transformers.html). The same logic 
 can be used for internal transformers.
 
+Testing
+--------
+See ``tests/README.rst`` file for more information on testing.
+
 Other contributions
 -------------------
 All contributions are welcome. If you're changing the source code remember to add tests - see `tests/README.md` for more information.
```
Comment (mnojek) [inline CONTRIBUTING.md:68]:
```suggestion
All contributions are welcome. If you're changing the source code remember to add tests - see `tests/README.rst` for more information.
```

**Review by mnojek — APPROVED** [review]

**Pull request merged**